src/psyclone/parse/kernel.py

```diff
@@ -534,27 +538,44 @@ def get_procedure(ast, name, modast):
                             "Kernel type {0} binds to a specific procedure but"
                             " does not use 'code' as the generic name.".
                             format(name))
-                    bname = statement.bname
+                    bname = statement.bname.lower()
                 else:
-                    bname = statement.name
+                    bname = statement.name.lower()
                 break
         if bname is None:
-            raise ParseError(
-                "Kernel type {0} does not bind a specific procedure".
-                format(name))
-        if bname == '':
+            # If no type-bound procedure found, search for an explicit
+            # interface that has module procedures.
+            bname, subnames = get_kernel_interface(name, modast)
+            if bname is None:
+                # no interface found either
+                raise ParseError(
+                    "Kernel type {0} does not bind a specific procedure or \
+                    provide an explicit interface".format(name))
+        elif bname == '':
             raise InternalError(
                 "Empty Kernel name returned for Kernel type {0}.".format(name))
-        code = None
-        for statement, _ in fpapi.walk(modast, -1):
-            if isinstance(statement, fparser1.block_statements.Subroutine) \
-               and statement.name == bname:
-                code = statement
-                break
-        if not code:
-            raise ParseError(
-                "kernel.py:KernelProcedure:get_procedure: Kernel subroutine "
-                "'{0}' not found.".format(bname))
+        # add the name of the tbp to the list of strings to search for.
```

Comment:
Please could you move this comment into the branch to which it applies (i.e. move to line 559).

Reply:
Done.